### PR TITLE
remove all tmp dir after test

### PR DESCRIPTION
**Pull request by guangxuli**
Remove all tmp directory and add a break is better.

### CODE REVIEW

File: test/integration/shortcut_expansion_test.go [hunk 1]
```diff
@@ -38,6 +38,7 @@ func TestCachingDiscoveryClient(t *testing.T) {
 	if err != nil {
 		t.Fatalf("unexpected error: %v", err)
 	}
+	defer os.RemoveAll(cacheDir)
```
Comment (deads2k) [inline test/integration/shortcut_expansion_test.go]: Only remove this if the test is successful.

**Comment by guangxuli** [comment]
@deads2k do you mean we need check the tmp file/dir created at the beginning of test when the test is fail?

**Comment by deads2k** [comment]
> @deads2k do you mean we need check the tmp file/dir created at the beginning of test when the test is fail?

Use https://godoc.org/testing#T.Failed in your `defer` func

**Comment by guangxuli** [comment]
@deads2k PTAL, thanks.

**Comment by deads2k** [comment]
lgtm.  @mfojtik I'm not sure the status of the merge queue.

**Comment by mfojtik** [comment]
lgtm [merge]

**Pull request merged**